pywb/static/wombat.js

```diff
@@ -1326,6 +1330,144 @@ var _WBWombat = function($wbwindow, wbinfo) {
     }
 
     //============================================
+    function initPreserveWorker() {
+        if (!wbUsePresWorker) {
+            return;
+        }
+
+        var Preserver = (function(Worker) {
+            function PWorker(prefix, mod) {
+                if (!(this instanceof PWorker)) {
+                    return new PWorker(prefix, mod);
+                }
+                if ($wbwindow === $wbwindow.__WB_replay_top) {
+                    // we are top and can will own this worker
+                    // setup URL for the kewl case
+                    var workerURL = wbinfo.static_prefix +
+                        'wombatPreservationWorker.js?prefix=' +
+                        encodeURIComponent(prefix) + '&mod=' +
+                        encodeURIComponent(mod);
+                    this.worker = new Worker(workerURL);
+                    if (typeof $wbwindow.URL === 'undefined') {
+                        // not kewl case so we must init the worker ourselves :'(
+                        this.worker.postMessage({
+                            'type': 'init',
+                            'prefix': prefix,
+                            'mod': mod,
+                        });
+                    }
+                } else {
+                    this.worker = null;
+                }
+            }
+
+            PWorker.prototype.deferredSheetExtraction = function(rules) {
+                // if no rules this a no op
+                if (rules.length === 0) return;
+                function extract() {
+                    // loop through each rule of the stylesheet
+                    var media = [];
+                    for (var j = 0; j < rules.length; ++j) {
+                        var rule = rules[j];
+                        if (rule instanceof CSSMediaRule) {
+                            // we are a media rule so get its text
+                            media.push(rule.cssText);
+                        }
+                    }
+                    if (media.length > 0) {
+                        // we have some media rules to preserve
+                        WBPreserWorker.preserveMedia(media);
+                    }
+                }
+                // depending on if we have promises, defer things until
+                // next time the Promise.resolve or setTimeout Qs are cleared
+                if (typeof $wbwindow.Promise !== 'undefined') {
```

Comment:
Is it safe to assume we have Promises if we have workers? If so, maybe should just do it one or the other, as there's already enough conditionals here. :) I think we use Promises already in wombat..